bin/cassandra.in.sh

```diff
@@ -84,6 +84,12 @@ JAVA_AGENT="$JAVA_AGENT -javaagent:$CASSANDRA_HOME/lib/jamm-0.3.2.jar"
 # Added sigar-bin to the java.library.path CASSANDRA-7838
 JAVA_OPTS="$JAVA_OPTS:-Djava.library.path=$CASSANDRA_HOME/lib/sigar-bin"
 
+operating_system=$(uname -s | tr '[:upper:]' '[:lower:]')
+corretto_lib_pattern="AmazonCorrettoCryptoProvider-*-$operating_system-$(uname -m).jar"
+corretto_lib="$(find "${CASSANDRA_HOME}"/lib/corretto -type f -name "$corretto_lib_pattern" -exec basename {} \; | head -n 1)"
+if [ -f "${CASSANDRA_HOME}/lib/corretto/${corretto_lib}" ]; then
+  CLASSPATH="$CLASSPATH${CASSANDRA_HOME}/lib/corretto/$corretto_lib";
+fi
```

Comment:
maybe an else to print a warning that the other architetcure (e.g. MIPS, power. or whatever) is not supported

Reply:
they are supported in the sense that it just falls back to normal provider.
i'm guessing such feedback is provided from java logging at startup …?

Reply:
yes it is logged